__test__/index.test.js

```diff
@@ -0,0 +1,43 @@
+const chai = require("chai");
+const expect = chai.expect;
+const GRAPHQL_ROUTE = "/the-graph-badges";
+const url = `http://127.0.0.1:8000/subgraphs/name/badgeth-dao`;
+const request = require("supertest")(url);
+
+describe("GraphQL", () => {
+  it("Returns expected badge definitions", (done) => {
+    request
+      .post(GRAPHQL_ROUTE)
+      .send({ query: "{ badgeDefinitions(first: 3, orderBy: id) { id } }" })
+      .expect(200)
+      .end((err, res) => {
+        if (err) return done(err);
+        const { data } = res.body;
+        const { badgeDefinitions } = data;
+
+        expect(badgeDefinitions[0].id).to.equal("28 Epochs Later");
+        expect(badgeDefinitions[1].id).to.equal("Delegation Nation");
+        expect(badgeDefinitions[2].id).to.equal("Delegation Streak");
```

Comment:
interesting results here, this one doesn't pass until the subgraph is completely indexed

Reply:
Hmm - that's an interesting one - I <3 the unit testing, but I'm worried about tests that block us until indexing has fully happened. I've read some evolving best practices in this thread: https://forum.thegraph.com/t/testing-best-practices/903/15

Also worth checking out https://github.com/LimeChain/subtest another MVP in the works project

Reply:
It always sucks when CI/CD takes 20 minutes - maybe that's unavoidable tho

Reply:
I think it will be useful to have as a smoke test post deployment, to make sure everything is in place. If it takes a while to fully index everything, it shouldn't be a part of ci/cd, at least not for the moment. We could also write some simpler tests that simply check that the subgraph is up there, deployed correctly, not something like this that's hard- coded